tests/test_printer.py

```diff
@@ -382,13 +382,23 @@ class PlusCustomFormatOp(Operation):
 
     @classmethod
     def parse(cls, result_types: List[Attribute],
-              parser: Parser) -> PlusCustomFormatOp:
-        lhs = parser.parse_ssa_value()
-        parser.skip_white_space()
-        parser.parse_char("+")
-        rhs = parser.parse_ssa_value()
-        return PlusCustomFormatOp.create(operands=[lhs, rhs],
-                                         result_types=result_types)
+              parser: BaseParser) -> PlusCustomFormatOp:
+
+        def get_ssa_val(name: Span) -> SSAValue:
+            if name.text not in parser.ssaValues:
+                parser.raise_error('SSA Value used before assignment', name)
+            return parser.ssaValues[name.text]
+
+        lhs = parser.expect(parser.try_parse_value_id,
```

Comment:
I would prefer not letting the user call `parser.expect`, and instead define `parser.parse_value_id`.